.github/workflows/main.yml

```diff
@@ -0,0 +1,48 @@
+name: "Smart Contracts Unit Tests"
+on:
+   push:
+    branches: [main]
+   pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [main]
+
+jobs:
+  unit-test:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Use Node 16
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Tests
+        run: npm run test
+
+  deploy:
```

Comment:
I would remove the whole deploy task and instead would ad a step like:
- name: Deploy
        run: npm run deploy hardhat
        
 As it's currently creating another docker image just for the deploy and installing the dependecies

Reply:
done